src/libs/Middleware/Pagination.ts

```diff
@@ -0,0 +1,132 @@
+import fastMerge from 'expensify-common/dist/fastMerge';
+import type {OnyxCollection} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import type {ApiCommand} from '@libs/API/types';
+import Log from '@libs/Log';
+import PaginationUtils from '@libs/PaginationUtils';
+import CONST from '@src/CONST';
+import type {OnyxCollectionKey, OnyxPagesKey, OnyxValues} from '@src/ONYXKEYS';
+import type {Request} from '@src/types/onyx';
+import type {PaginatedRequest} from '@src/types/onyx/Request';
+import type Middleware from './types';
+
+type PaginationConfig<TResourceKey extends OnyxCollectionKey, TPageKey extends OnyxPagesKey> = {
+    initialCommand: ApiCommand;
+    previousCommand: ApiCommand;
+    nextCommand: ApiCommand;
+    resourceCollectionKey: TResourceKey;
+    pageCollectionKey: TPageKey;
+    sortItems: (items: OnyxValues[TResourceKey]) => Array<OnyxValues[TResourceKey] extends Record<string, infer TResource> ? TResource : never>;
```

Comment:
Interesting suggestion. For now there aren't any resources that come presorted from the back-end AFAIK, so if/when we change that would be a better time to make it optional imo. I'm glad you're thinking about how this might be re-used for report pagination though.